mineral-ui/mineral-ui
# Extract mineral-ui-icons package

### Description

Extracts pre-built icons into separate `mineral-ui-icons` package

* Updates node version to latest LTS
* Splits build script into `build:package` and `build:website`
* Refactors `build:icons` script to use async/await, to process any svg's, and copy internal mineral icons back to the Icon component.
* Resolves icon paths in different environments using `babel-plugin-module-resolver` and flow mappings
* Updates documentation in website for importing icons
* Recreates async Icon category demo

### Motivation and context

* speed up build
* make custom mineral icons publicly available
* lessen likelihood that users will accidentally import all of the icons
* establish a workflow for adding custom public mineral icons
* establish a pattern for how to manage "internal" mineral icons

closes #315 
close #365 

### Screenshots, videos, or demo, if appropriate

https://315-mineral-ui-icons--mineral-ui.netlify.com/

https://www.npmjs.com/package/mineral-ui-icons

### How to test

* How to check if working in mineral-ui properly
  * Website
    * All Icons work, regular and those in MenuItem
    * `npm run build:analyze` still splits code properly
    * Icon category demo renders properly and is still async
  * Components
    * build and check commonjs src of component using an icon, eg. `MenuItem` - should be transpiled to ./Icon.  It should not use `mineral-ui-icons` as this would require a peerDependency.
* How to test if working in consumer app
  * I published a beta version of the `mineral-ui-icons` package, so you can just install that in CRA and see if it works.  Otherwise, you can build a version locally, and install the .tar file in CRA to test.

### Types of changes

- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

Breaking feature!
* Breaking in that users must now opt-in to the pre-built icons by installing the `mineral-ui-icons` npm package.
* Several new icons available - `IconDanger`, `IconSuccess`, `IconWarning`

### Checklist
